tests/unit/delegate.ts

```diff
@@ -0,0 +1,180 @@
+import registerSuite = require('intern!object');
+import assert = require('intern/chai!assert');
+import { emit } from 'dojo-core/on';
+import { Handle } from 'dojo-core/interfaces';
+import delegate from 'src/delegate';
+import * as dom from 'src/dom';
+
+let container = document.createElement('div');
+let handles: (Handle)[] = [];
+
+function testDelegate(...args: any[]) {
+	let handle = delegate.apply(null, arguments);
+	handles.push(handle);
+	return handle;
+}
+
+registerSuite({
+	name: 'delegate',
+
+	setup() {
+		document.body.appendChild(container);
+	},
+
+	teardown() {
+		document.body.removeChild(container);
+	},
+
+	afterEach() {
+		while (handles.length > 0) {
+			handles.pop().destroy();
+		}
+		container.innerHTML = '';
+	},
+
+	'CSS selector matching multiple elements'() {
+		let called = 0;
+		let buttonOne = document.createElement('button');
+		let buttonTwo = document.createElement('button');
+
+		delegate(container, 'button', 'click', function() {
+			called++;
+		});
+
+		container.appendChild(buttonOne);
+		container.appendChild(buttonTwo);
+
+		buttonOne.click();
+		buttonTwo.click();
+		assert.strictEqual(called, 2);
+	},
+
+	'CSS selector with array of events'() {
+		let called = 0;
+		let button = document.createElement('button');
+
+		delegate(container, 'button', ['click'], function() {
+			called++;
+		});
+
+		container.appendChild(button);
+
+		button.click();
+		assert.strictEqual(called, 1);
+	},
+
+	'CSS selector and text node event target'() {
+		let called = 0;
+		let div = document.createElement('div');
+		div.innerHTML = 'test';
+
+		testDelegate(container, 'div', 'click', function() {
+			called++;
+		});
+
+		container.appendChild(div);
+		
+		emit(div, {
+			type: 'click',
+			bubbles: true,
+			cancelable: true
+		});
+
+		assert.strictEqual(called, 1);
+	},
+
+	'Listening on body'() {
+		let called = 0;
+		let button = document.createElement('button');
+
+		testDelegate(document.body, 'button', 'click', function() {
+			called++;
+		});
+
+		document.body.appendChild(button);
```

Comment:
This isn't getting cleaned up.  I'm changing it to add to `container` since that should still effectively test it.